docs/tutorials/building-a-python-source.md

```diff
@@ -0,0 +1,196 @@
+# How to Build a Python Source
+
+## Summary
+
+This article provides a checklist for how to create a python source. Each step in the checklist has a link to a more detailed explanation below.
+
+## Requirements
+
+Docker, Python, and Java with the versions listed in the [tech stack section](../architecture/tech-stack.md).
+
+{% hint style="warn" %}
+All the commands below assume that `python` points to a version of python 3. On some systems, `python` points to a Python2 installation and `python3` points to Python3. If this is the case on your machine, substitute all `python` commands in this guide with `python3` . Otherwise, make sure to install Python 3 before beginning.
+{% endhint %}
+
+## Checklist
+
+### Creating a Source
+* Step 1: Create the source using template
+* Step 2: Build the newly generated source `./gradlew :airbyte-integrations:connectors:source-<source-name>:build`
+* Step 3: Set up your Airbyte development environment 
+* Step 4: Implement `spec` (and define the specification for the source `airbyte-integrations/connectors/source-<source-name>/spec.json`)
+* Step 5: Implement `check`
+* Step 6: Implement `discover`
+* Step 7: Implement `read`
+* Step 8: Set up Standard Tests
+* Step 9: Write unit tests or integration tests
+* Step 10: Update the `README.md` (If API credentials are required to run the integration, please document how they can be obtained or link to a how-to guide.)
+* Step 11: Add the connector to the API/UI (by adding an entry in `airbyte-config/init/src/main/resources/seed/source_definitions.yaml`)
+* Step 12: Add docs (in `docs/integrations/sources/<source-name>.md`)
+
+{% hint style="info" %}
+Each step of the Creating a Source checklist is explained in more detail below.
+{% endhint %}
+
+{% hint style="info" %}
+All `./gradlew` commands must be run from the root of the airbyte project.
+{% endhint %}
+
+### Submitting a Source to Airbyte 
+* Submit a PR.
+* Coordinate with an Airbyte engineer (via the PR) to add test credentials so that we can run tests for the integration in the CI. (We will create our own test account once you let us know what source we need to create it for.)
```

Comment:
```suggestion
* To run integration tests, Airbyte needs access to a test account/environment. Coordinate with an Airbyte engineer (via the PR) to add test credentials so that we can run tests for the integration in the CI. (We will create our own test account once you let us know what source we need to create it for.)
```